scope_registry.go

```diff
@@ -86,19 +86,27 @@ func (r *scopeRegistry) Subscope(parent *scope, prefix string, tags map[string]s
 		return NoopScope.(*scope)
 	}
 
-	key := scopeRegistryKey(prefix, parent.tags, tags)
-
+	preSanitizeKey := keyForPrefixedStringMapsAsKey(prefix, parent.tags, tags)
 	r.mu.RLock()
-	if s, ok := r.lockedLookup(key); ok {
+	if s, ok := r.lockedLookup(preSanitizeKey.CastAsString()); ok {
 		r.mu.RUnlock()
```

Comment:
Just a random thought and should not be fixed here, but would it make sense to use sync.Map here? It avoids the mutex locking, when the key is already present (and just use an atomic value lookup) and thus you don't need to remember the unlocking here.

Reply:
+1. We could do this optimization as a follow up.